client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -8,10 +9,136 @@
+        `substance_painter.project.ProjectWorkflow.Default`
+
+    Args:
+        substance_attr (str): The `substance_painter.project` attribute,
+            for example "ProjectWorkflow.Default"
+
+    Returns:
+        Any: Substance Python API object of the project attribute.
+
+    Raises:
+        ValueError: If attribute does not exist on the
+            `substance_painter.project` python api.
+    """
+    root = substance_painter.project
+    for attr in substance_attr.split("."):
+        root = getattr(root, attr, None)
+        if root is None:
+            raise ValueError(
+                "Substance Painter project attribute"
+                f" does not exist: {substance_attr}")
+
+    return root
+
+
+def get_template_by_name(name: str, templates: list[dict]) -> dict:
+    return next(
+        template for template in templates
+        if template["name"] == name
+    )
+
+
+class SubstanceProjectConfigurationWindow(QtWidgets.QDialog):
+    """The pop-up dialog allows users to choose material
+    duplicate options for importing Max objects when updating
+    or switching assets.
+    """
+    def __init__(self, project_templates):
+        super(SubstanceProjectConfigurationWindow, self).__init__()
+        self.setWindowFlags(self.windowFlags() | QtCore.Qt.FramelessWindowHint)
+
+        self.import_cameras = False
+        self.preserve_strokes = False
+        self.template_name = None
+        self.template_names = [template["name"] for template
+                               in project_templates]
+        self.project_templates = project_templates
+
+        self.widgets = {
+            "label": QtWidgets.QLabel(
+                "Select your template for project configuration"),
+            "template_options": QtWidgets.QComboBox(),
+            "import_cameras": QtWidgets.QCheckBox("Import Cameras"),
+            "preserve_strokes": QtWidgets.QCheckBox("Preserve Strokes"),
+            "clickbox": QtWidgets.QWidget(),
+            "combobox": QtWidgets.QWidget(),
+            "buttons": QtWidgets.QWidget(),
+            "okButton": QtWidgets.QPushButton("Ok"),
+            "cancelButton": QtWidgets.QPushButton("Cancel")
+        }
+        for template in self.template_names:
+            self.widgets["template_options"].addItem(template)
+
+        template_name = self.widgets["template_options"].currentText()
+        self.get_boolean_setting(template_name)
+        # Build clickboxes
+        layout = QtWidgets.QHBoxLayout(self.widgets["clickbox"])
+        layout.addWidget(self.widgets["import_cameras"])
+        layout.addWidget(self.widgets["preserve_strokes"])
+        # Build combobox
+        layout = QtWidgets.QHBoxLayout(self.widgets["combobox"])
+        layout.addWidget(self.widgets["template_options"])
+
+        # Build buttons
+        layout = QtWidgets.QHBoxLayout(self.widgets["buttons"])
+        layout.addWidget(self.widgets["okButton"])
+        layout.addWidget(self.widgets["cancelButton"])
```

Comment:
You could also use a [`QDialogButtonBox`](https://doc.qt.io/qt-5/qdialogbuttonbox.html)? 

Something along the lines of
```python
buttons = QtWidgets.QDialogButtonBox()
buttons.add(QtWidgets.QDialogButtonBox.Ok)
buttons.add(QtWidgets.QDialogButtonBox.Cancel)
```

I believe in a `QDialog` these also directly register as the default accept and cancel buttons so that you can list to the actual dialog's `QDialog.accepted` and `QDialog.rejected` signals. That way you also respond to e.g. a user clicking escape to close the dialog?

Reply:
Let's me try to see if this is possible for implementing `dialog.accepted` and `dialog.rejected` in prompt along with `QtWidgets.QDialogButtonBox()`